cpp/tests/test_ide_secir.cpp

```diff
@@ -489,6 +490,36 @@ TEST(IdeSecir, testModelConstraints)
     constraint_check = model.check_constraints(dt);
     EXPECT_TRUE(constraint_check);
 
+    // --- Correct wrong setup so that next check can be tested.
+    // Create TimeSeries with num_transitions elements.
+    mio::TimeSeries<ScalarType> init_enough_timepoints2(num_transitions);
+    // Add time points for initialization of transitions.
+    init_enough_timepoints2.add_time_point(-5, vec_init);
+    while (init_enough_timepoints2.get_last_time() < 0) {
+        init_enough_timepoints2.add_time_point(init_enough_timepoints2.get_last_time() + dt, vec_init);
+    }
+
+    // Set transitions again to TimeSeries with original last time point.
+    model.m_transitions = init_enough_timepoints2;
+
+    // Check that constraints are fulfilled.
+    constraint_check = model.check_constraints(dt);
+    EXPECT_FALSE(constraint_check);
+
+    // --- Test with TimeSeries for populations that contains more than one time point.
+    // Create TimeSeries with num_compartments elements.
+    mio::TimeSeries<ScalarType> populations_two_timepoints(num_compartments);
+    // Add time points.
+    populations_two_timepoints.add_time_point(-2, vec_init);
+    while (populations_two_timepoints.get_last_time() < 0) {
+        populations_two_timepoints.add_time_point(populations_two_timepoints.get_last_time() + dt, vec_init);
+    }
+
+    model.m_populations = populations_two_timepoints;
+
+    constraint_check = model.check_constraints(dt);
+    EXPECT_TRUE(constraint_check);
+
```

Comment:
I would change the order a little bit: there should be only one case where we have right conditions and an EXPECT_FALSE statement (see line 464-477). This should also be the last test. For the test with the wrong number of time points in m_populations you should just match the last time point of m_transitions from the last test (here 1, i guess) and not correct all consitions in between.